Operators/Types/lib/io/video/SpoutInput.cs

```diff
@@ -0,0 +1,329 @@
+                Format textureFormat = Format.B8G8R8A8_UNorm;
+                if (GetSenderInfo(ref senderWidth, ref senderHeight, ref senderHandle, ref directXFormat))
+                {
+                    textureFormat = (Format) directXFormat;
+                }
+
+                // create several textures with a given format with CPU access
+                // to be able to read out the initial texture values
+                if (ImagesWithGpuAccess.Count == 0
+                    || ImagesWithGpuAccess[0].Description.Format != textureFormat
+                    || ImagesWithGpuAccess[0].Description.Width != (int)width
+                    || ImagesWithGpuAccess[0].Description.Height != (int)height
+                    || ImagesWithGpuAccess[0].Description.MipLevels != 1)
+                {
+                    var imageDesc = new Texture2DDescription
+                    {
+                        BindFlags = BindFlags.ShaderResource,
+                        Format = textureFormat,
+                        Width = (int)width,
+                        Height = (int)height,
+                        MipLevels = 1,
+                        SampleDescription = new SampleDescription(1, 0),
+                        Usage = ResourceUsage.Default,
+                        OptionFlags = ResourceOptionFlags.Shared,
+                        CpuAccessFlags = CpuAccessFlags.None,
+                        ArraySize = 1
+                    };
+
+                    DisposeTextures();
+
+                    Log.Debug($"Spout input wxh = {senderWidth}x{senderHeight}, " +
+                              $"handle = {senderHandle}, " +
+                              $"format = {textureFormat} ({directXFormat})");
+
+                    for (var i = 0; i < NumTextureEntries; ++i)
+                    {
+                        ImagesWithGpuAccess.Add(new Texture2D(device, imageDesc));
+                    }
+                    _width = width;
+                    _height = height;
+
+                    _currentIndex = 0;
+                }
+
+                // sanity check
+                if (_spoutDX == null || width == 0 || height == 0 || _width != width || _height != height)
+                    return false;
+
+                // copy the spout texture to an internal image
+                if (_spoutDX.IsFrameNew)
+                {
+                    var immediateContext = device.ImmediateContext;
+                    var readableImage = ImagesWithGpuAccess[_currentIndex];
+
+                    Texture2D readTexture = new Texture2D(_spoutDX.SenderTexture.__Instance);
+                    immediateContext.CopyResource(readTexture, readableImage);
+                    _currentIndex = (_currentIndex + 1) % NumTextureEntries;
+
+                    Texture.Value = readableImage;
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Debug("Spout input texture reception failed: " + e.ToString());
```

Comment:
```suggestion
                Log.Debug("Spout input texture reception failed: " + e.ToString(), SymbolChildId);
```
This would highlight / center the operator when clicking on the message in the log window.